### ISSUE TITLE
Ignored code won't print comments inside

### ISSUE THREAD

**Issue by cavanmflynn** [prettier-solidity/prettier-plugin-solidity#418]
This contract:

```solidity
pragma solidity >=0.8.0 <0.9.0;

// prettier-ignore
contract Example {
    // Test comment
}
```

will throw the following error:

```
Error: Comment "Test comment" was not printed. Please report this error!
```

**Comment by fvictorio** [comment]
Oh damn, this is a hard one. The straightforward solution is to make the comment extractor aware of `//prettier-ignore` comments. Since the extractor is ours and we are the only users, it's not a *terrible* idea, but it would be a pity to make it less reusable.

Maybe the extractor could receive certain tokens that enable/disable it? For example:

```js
const commentsExtractor = new SolidityComments({
  ignoreAfter: ["// prettier-ignore"]
})
```

**Comment by Kelzcope** [comment]
Pull request comand to pull code

**Comment by fvictorio** [comment]
@Kelzcope what?

**Comment by Janther** [comment]
maybe we can solve this on printing. If we find a `prettier-ignore` belonging to a block then we automatically scan through all the comments within the rang of the block and mark them as printed.

**Comment by Janther** [comment]
I'd do it but I'll be offline next week maybe if nothing has happened I can give this a try.

**Comment by Janther** [comment]
This was solved in #428
